File: 1127-last-stone-weight/1127-last-stone-weight.cpp
class Solution {
public:
    int lastStoneWeight(vector<int>& stones) {
         priority_queue<int>pq;
        for(int s : stones)
            pq.push(s);
        while(pq.size()>=2){
            int y = pq.top();
            pq.pop();
            int x = pq.top();
            pq.pop();
            if(y>x){
                pq.push(y-x);
            }
        }
        return !pq.empty()?pq.top():0;
    }
};

File: 1127-last-stone-weight/1127-last-stone-weight.java
class Solution {
    public int lastStoneWeight(int[] stones) {
        PriorityQueue<Integer>pq = new PriorityQueue<Integer>(Collections.reverseOrder());
        for(int s : stones)
            pq.add(s);
        while(pq.size()>=2){
            int y = pq.poll();
            int x = pq.poll();
            if(y>x){
                pq.add(y-x);
            }
        }
        return pq.peek()!=null?pq.peek():0;
    }
}